schema/oneof_test.go

```diff
@@ -23,6 +23,20 @@ type oneOfTestObjectA struct {
 	S any `json:"s"`
 }
 
+type oneOfTestInlineObjectB struct {
+	Message string `json:"message"`
+	Choice  string `json:"choice"`
+}
+
+func (o oneOfTestInlineObjectB) String() string {
+	return o.Message
+}
+
+type oneOfTestInlineObjectC struct {
+	M      string `json:"m"`
+	Choice string `json:"choice"`
+}
```

Comment:
Wouldn't it make for a better test if `M` were _not_ a string, and, perhaps, if "C" did not have the same number of fields as "B"?